pkg/util/util_windows.go

```diff
@@ -29,19 +29,27 @@ func Umask(mask int) (int, error) {
 }
 
 func CleanupSocketFile(socketPath string) error {
-	if _, err := os.Lstat(socketPath); err != nil {
-		// If the file does not exist, then the cleanup can be considered successful.
-		if os.IsNotExist(err) {
-			return nil
+	socketExists, err := DoesSocketExist(socketPath)
+	if err != nil {
+		return err
+	}
+	if socketExists {
+		// TODO: Until the bug - https://github.com/golang/go/issues/33357 is fixed, os.stat wouldn't return the
```

Comment:
I think this comment fits better under `DoesSocketExist` to describe why we're not checking the file mode.